### ISSUE TITLE
`await-async-events` doesn't seem to check calls from user sessions returned by userEvent.setup()

### ISSUE THREAD

**Issue by julienw** [testing-library/eslint-plugin-testing-library#812]
### What rule do you want to change?

await-async-events

### Does this change cause the rule to produce more or fewer warnings?

More warnings

### How will the change be implemented?

The code needs to detect calls to `userEvent.setup()` and calls to the resulting object's methods.

### Example code

```js
import userEvent from '@testing-library/user-event';
const user = userEvent.setup();
user.click(domElement);
```

### How does the current rule affect the code?

`user.click()` and others returns a `Promise` and should be `await`ed.

### How will the new rule affect the code?

```js
import userEvent from '@testing-library/user-event';
const user = userEvent.setup();
await user.click(domElement);
```

### Anything else?

see also https://github.com/testing-library/eslint-plugin-testing-library/issues/800

### Do you want to submit a pull request to change the rule?

No

**Comment by Kvanttinen** [comment]
Any estimate for this? We're about to start using this package in our project and this is a blocker for us.

**Comment by Belco90** [comment]
There is no ETA for this. PRs are more than welcome 🙂.

**Comment by Kvanttinen** [comment]
Ok, I was already looking for a solution for that yesterday, I'll see if I can come up with something.

**Comment by gerardo-rodriguez** [comment]
I'd love to see this as well. Are there any more recent updates on this? Thanks, everyone! 😄

**Comment by Stan-Stani** [comment]
This makes me think I shouldn't be migrating my code to do: `const user = userEvent.setup()` because I miss out on the rule catching stuff here.
